src/test/scala/circtTests/ConventionSpec.scala

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package circtTests
+
+import chisel3._
+import circt.convention
+import circt.stage.ChiselStage
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+
+class ConventionSpec extends AnyFunSpec with Matchers {
+  private class PassThrough extends Module {
+    val i = IO(Input(new Bundle {
+      val x = UInt(16.W)
+      val y = UInt(16.W)
+    }))
+    val o = IO(Output(new Bundle {
+      val x = UInt(16.W)
+      val y = UInt(16.W)
+    }))
+    o := i
+  }
+
+  val stage = new ChiselStage
+
+  describe("convention.scalarized") {
+    it("should scalarize the ports of PassThrough 'a'") {
+    
+    val expected =
+    """|module PassThrough(
+       |input         clock,
+       |              reset,
+       | input  [15:0] i_x,
+       |               i_y,
+       | output [15:0] o_x,
+       |               o_y
+       |);
+       |  assign o_x = i_x;
+       |  assign o_y = i_y;
+       |endmodule
+    """.stripMargin.split("\n").map(_.trim)
```

Comment:
This test can be tightened up to only check the scalarized ports as opposed to everything.

Changing the `ChiselStage` invocation to use `--loweringOptions=disallowPortDeclSharing` will allow more exact checking of the ports.

Reply:
ah this is much better, thanks.